Doc/whatsnew/3.10.rst

```diff
@@ -166,6 +166,48 @@ See :class:`typing.Callable`, :class:`typing.ParamSpec`,
 
 (Contributed by Ken Jin in :issue:`41559`.)
 
+Better error messages in the parser
+-----------------------------------
+
+When parsing code that contains unclosed parentheses or brackets the interpreter
+now includes the location of the unclosed bracket of parentheses instead of displaying
+*SyntaxError: unexpected EOF while parsing* or pointing to some incorrect location.
+For instance, consider the following code (notice the unclosed '{'):
+
+.. code-block:: python
+
+    expected = {9: 1, 18: 2, 19: 2, 27: 3, 28: 3, 29: 3, 36: 4, 37: 4,
+                38: 4, 39: 4, 45: 5, 46: 5, 47: 5, 48: 5, 49: 5, 54: 6,
+    some_other_code = foo()
+
+previous versions of the interpreter reported confusing places as the location of
+the syntax error:
+
+.. code-block:: text
+
+      File "example.py", line 3
+        some_other_code = foo()
+                        ^
+    SyntaxError: invalid syntax
```

Comment:
The indentation here is slightly strange, here's how it looks:
![ex_105371672-b04f5a80-5bb9-11eb-91cb-edebbb60f705](https://user-images.githubusercontent.com/28750310/105372543-8cd8df80-5bba-11eb-9253-e17e730571dd.png)


```suggestion
   File "example.py", line 3
       some_other_code = foo()
                       ^
   SyntaxError: invalid syntax
```

Reply:
Thanks for the suggestion!